cross-root-aria-reflection.md

```diff
@@ -116,6 +116,400 @@ Or even further by accepting a token list on `reflects` or a similarly shaped at
+relationships begin to change or become impossible. To suppor this, we've assumed the presence of
+the ARIA Attribute Delegation API in following code examples:
+
+```html
+<x-label id="cb1-label">State</x-label>
+<x-combobox>
+    <x-input-group>
+        <x-input
+            aria-labeledby="cb1-label"
+            role="combobox"
+            aria-autocomplete="list"
+            aria-expanded="true"
+            aria-controls="cb1-listbox"
+            aria-activedescendant="lb1-ak"
+        >
+            #shadow-root delegates="aria-labeledby role aria-autocomplete aria-expanded aria-controls aria-activedescendant"
+                <input
+                    type="text"
+                    auto-role
+                    auto-aria-labeledby
+                    auto-aria-autocomplete
+                    auto-aria-expanded
+                    auto-aria-controls
+                    auto-aria-activedescendant
+                />
+        </x-input>
+        <x-button
+            aria-labeledby="cb1-label"
+            tabindex="-1"
+            aria-expanded="true"
+            aria-controls="cb1-listbox"
+        ></x-button>
+    </x-input-group>
+    <x-listbox
+        aria-labeledby="cb1-label"
+        options='[["Alabama", "lb1-al"], ["Alaska", "lb1-ak"]]'
+    ></x-listbox>
+</x-combobox>
+```
+
+Here we've moved from the `for` attribute on a `<label>` element to giving its host an ID so that
+other elements can reference it via `aria-labelledby`. This persists the accessible relationship, 
+but it removes the previously managed interactions, like clicking the `<label>` focusing on the
+input. We also see the `aria-activedescendant` relationship broken as the ID `lb1-ak` moves into 
+the shadow root of the `<x-listbox>` element. This is the first place the ARIA Attribute Reflection 
+benefits the refactor of the pattern from raw DOM to custom elements.
+
+```html
+<x-label id="cb1-label">State</x-label>
+<x-combobox>
+    <x-input-group>
+        <x-input
+            aria-labeledby="cb1-label"
+            role="combobox"
+            aria-autocomplete="list"
+            aria-expanded="true"
+            aria-controls="cb1-listbox"
+            aria-activedescendant="lb1-ak"
+        >
+            #shadow-root delegates="aria-labeledby role aria-autocomplete aria-expanded aria-controls aria-activedescendant"
+                <input
+                    type="text"
+                    auto-role
+                    auto-aria-labeledby
+                    auto-aria-autocomplete
+                    auto-aria-expanded
+                    auto-aria-controls
+                    auto-aria-activedescendant
+                />
+        </x-input>
+        <x-button
+            aria-labeledby="cb1-label"
+            tabindex="-1"
+            aria-expanded="true"
+            aria-controls="cb1-listbox"
+        >
+            #shadow-root reflects="role"
+                <button reflect-role></button>
+        </x-button>
```

Comment:
Maybe I'm misunderstanding this example, but shouldn't the `<button>` be delegating its `role` to the host, rather than reflecting it? AIUI `role` would always be delegated, never reflected.

(`role` reflection also opens up its own set of questions about "behaviors" discussed elsewhere, e.g. the `<x-button>` should [handle spacebar/enter events correctly](https://adrianroselli.com/2022/04/brief-note-on-buttons-enter-and-space.html), but maybe that should be solved later. 🙂)

Reply:
This may belong on a different example, as I think you're right that this specific relationship doesn't need to be maintained in this interface.

Reflecting role from a button to a root is more for when a collection of buttons is gathered under a `role="radiogroup"` or `role="group"` parent and the `role="radio"` or `role="checkbox"` needs to be the direct descendent...opens up a different can of works, however.